lib/reactor/builder.ex

```diff
@@ -117,88 +96,67 @@ defmodule Reactor.Builder do
   def add_step(reactor, name, impl, arguments \\ [], options \\ [])
 
   def add_step(reactor, _name, _impl, _arguments, _options) when not is_reactor(reactor),
-    do: {:error, ArgumentError.exception("`reactor`: not a Reactor")}
+    do: {:error, argument_error(:reactor, "not a Reactor")}
 
   def add_step(_reactor, _name, _impl, arguments, _options) when not is_list(arguments),
-    do: {:error, ArgumentError.exception("`arguments` is not a list")}
+    do: {:error, argument_error(:arguments, "not a list")}
```

Comment:
this pattern is nice, but we should include the value in there potentially, i.e `argument_error(:arguments, "not a list", arguments)` and it would add a `got: #{inspect(actual)}`